src/Compilers/CSharp/Portable/Syntax/SyntaxEquivalence.cs

```diff
@@ -110,11 +110,13 @@ private static bool AreEquivalentRecursive(GreenNode before, GreenNode after, Fu
 
             if (topLevel)
             {
-                // Once we get down to the block level we don't need to go any further and we can
+                // Once we get down to the body level we don't need to go any further and we can
                 // consider these trees equivalent.
-                if ((SyntaxKind)before.RawKind == SyntaxKind.Block)
+                switch ((SyntaxKind)before.RawKind)
                 {
-                    return true;
+                    case SyntaxKind.Block:
```

Comment:
In general block doesn't mean a method body, is there a guarantee that we are always start "above" bodies?